### PR TITLE
[NFV] Fix NFV installation for SLE15

### PR DESCRIPTION
**Pull request by jlausuch**
Fix broken installation for 15-SP1 and other minor changes for 12-SP4

Proof runs:
12-SP4:
http://fromm.arch.suse.de/tests/1701
http://fromm.arch.suse.de/tests/1702

15-SP1:
http://fromm.arch.suse.de/tests/1703
http://fromm.arch.suse.de/tests/1704

### CODE REVIEW

**Comment by jlausuch** [comment]
@frankenmichl @cfconrad can you please review?

File: tests/kernel/mellanox_config.pm [hunk 1]
```diff
@@ -37,8 +37,9 @@ sub run {
     my $protocol = get_var('MLX_PROTOCOL') || 2;
 
     if (is_sle('>=15')) {
-        my $GA_REPO = 'http://download.suse.de/ibs/SUSE:/SLE-15:/GA/standard/SUSE:SLE-15:GA.repo';
-        zypper_call("ar -f -G $GA_REPO");
+        zypper_call("ar -f -G http://download.suse.de/ibs/SUSE:/SLE-15-SP1:/GA:/TEST/images/repo/SLE-15-SP1-Module-Development-Tools-POOL-x86_64-Media1/  dev_1");
```
Comment (frankenmichl) [inline tests/kernel/mellanox_config.pm]: Is it a good idea to hard-code the URLs for SLE15 SP1? For the next version you'll need to touch this again.

File: tests/nfv/prepare_env.pm [hunk 2]
```diff
@@ -43,48 +43,50 @@ sub run {
...
 
     record_info("INFO", "VSPerf Installation");
     assert_script_run("cd vswitchperf/systems");
+
+    # Hack to skip the OVS, DPDK and QEMU compilation as we will use the vanilla packages
+    assert_script_run("sed -n -e :a -e '1,8!{P;N;D;};N;ba' -i build_base_machine.sh");
     if (is_sle('>=15')) {
-        # Hack to skip the OVS, DPDK and QEMU compilation as we will use the vanilla packages
-        assert_script_run("sed -n -e :a -e '1,8!{P;N;D;};N;ba' -i build_base_machine.sh");
+        #zypper_call('--quiet in python2-pip');
```
Comment (frankenmichl) [inline tests/nfv/prepare_env.pm]: Please don't leave code commented out - noone will understand by reading the code thy this is commented..

File: tests/nfv/prepare_env.pm [hunk 3]
```diff
@@ -43,48 +43,50 @@ sub run {
...
     my $vsperf_version = get_required_var('VSPERF_VERSION');
-    my $dpdk_repo      = "http://dpdk.org/git/dpdk";
     my $vnf_image      = get_required_var('VNF_IMAGE');
     my $trafficgen_ip  = get_trafficgen_ip();
     my $children       = get_children();
     my $child_id       = (keys %$children)[0];
 
     record_info("INFO", "Install needed packages for NFV tests: OVS, DPKD, QEMU");
-    zypper_call('--quiet in git-core openvswitch-switch dpdk qemu tcpdump', timeout => 200);
+    zypper_call('--quiet in git-core openvswitch-switch dpdk qemu tcpdump', timeout => 60 * 4);
```
Comment (frankenmichl) [inline tests/nfv/prepare_env.pm]: I do get why you change timeouts to 60 * 4, but is there a reason to change the effective timeouts throughout the code?

File: tests/nfv/prepare_env.pm (hunk 2 above)
Comment (jlausuch) [reply tests/nfv/prepare_env.pm]: right, I forgot to remove it. Thanks for noticing

File: tests/nfv/prepare_env.pm (hunk 3 above)
Comment (jlausuch) [reply tests/nfv/prepare_env.pm]: some of them have been modified a bit, yes. The reason is timeouts in OSD again.. but in this concrete case I didn't want to write 60 * 3.33333  :)

File: tests/kernel/mellanox_config.pm (hunk 1 above)
Comment (jlausuch) [reply tests/kernel/mellanox_config.pm]: @asmorodskyi proposed so... you are right, but I might create another condition for SP2 ...

Comment (okurz) [reply tests/kernel/mellanox_config.pm]: I also recommend to strongly reconsider this version hardcoding. Yes, might be that we need to change it again but I suggest to start under the assumption that you will *not* need changes. I will merge the PR now nevertheless but this is something to consider for a followup task

**Pull request merged**